components/script/dom/formdata.rs

```diff
@@ -163,4 +164,35 @@ impl FormData {
 
         ret
     }
+
+    pub fn atoms(&self) -> Vec<Atom> {
+        let mut ret = vec![];
+        for (key, value) in self.data.borrow().iter() {
+            for _ in value {
+                ret.push(key.clone());
+            }
+        }
+
+        ret
+    }
+}
+
+impl Iterable for FormData {
+    type Key = USVString;
+    type Value = FileOrUSVString;
+
+    fn get_iterable_length(&self) -> u32 {
+        self.datums().len() as u32
+    }
+
+    fn get_value_at_index(&self, n: u32) -> FileOrUSVString {
+        match self.datums()[n as usize].value {
+            FormDatumValue::String(ref s) => FileOrUSVString::USVString(USVString(s.to_string())),
+            FormDatumValue::File(ref b) => FileOrUSVString::File(Root::from_ref(&*b)),
+        }
+    }
+
+    fn get_key_at_index(&self, n: u32) -> USVString {
+        USVString(self.atoms()[n as usize].clone().to_string())
+    }
```

Comment:
@nox Yes, they are really bad. And I think there are some similar issues should be fixed, too. (e.g. [components/script/dom/headers.rs](https://github.com/servo/servo/blob/d37d4d697ae8c8897384213be82f25a51a4aa835/components/script/dom/headers.rs#L253))